Python/signal_generation/receiver.py

```diff
@@ -1,78 +1,92 @@
-
-    def _calculate_travel_time(
-        self,
-        source_position: sg_pos.Position,
-        sound_speed: float = 1500,  # [m/s]
-    ) -> List[float]:
-        """Calculate the travel time from the source to the hydrophones.
-
-        Args:
-            source_position: Provide a position for the source.
-            sound_speed: The expected sound speed in the medium in meter per seconds.
-
-        Returns:
-            A numpy.array containing the travel times in milliseconds (ms).
-        """
-        distances = np.array([abs(pos - source_position) for pos in self.positions])
-        return distances / sound_speed * 10 ** 3
-
-    def generate_signals(
-        self,
-        output_length: float,  # [ms]
-        source: sg_source.Pinger,
-        source_position: sg_pos.Position,
-        amplitude: float = 1.0,
-        output_data_type: type = np.uint32,
-        sound_speed: float = 1500.0,
-    ) -> List[np.array]:
-        """Generate signals received by the hydrophones placed in the array.
-
-        Args:
-            output_length: The length of the expected output signal in milliseconds (ms).
-            source: Provide an initialized Pinger signal source.
-            source_position: Provide a position for the source.
-            amplitude: The scaling of the amplitudes of the pinger signal.
-            output_data_type: The datatype used for converting from float to integer. Must be
-                a numpy integer type.
-            sound_speed: The expected sound speed in the medium in meter per seconds.
-
-        Returns:
-            A list of numpy.arrays containing the received signal at the hydrophones
-            placed in the array.
-        """
-        travel_times = self._calculate_travel_time(
-            source_position=source_position,
-            sound_speed=sound_speed,
-        )
-
-        nr_of_samples = int(output_length * source.sampling_frequency)
-
-        signals = np.empty((len(self.positions), nr_of_samples), dtype=output_data_type)
-
-        for index, time in enumerate(travel_times):
-            converted_signal = sg_conv.convert_to_integer_type(
-                resulting_type=output_data_type,
-                input_signal=source.generate_signal(
-                    amplitude=amplitude,
-                    offset=time,
-                    length=output_length,
-                ),
+    """Simple container class to hold receiver position."""
+
+    positions: np.array
+
+    def __len__(self):
+        return len(self.positions)
+
+
+def calculate_transmission_loss_factors(
+    distances: np.array,
+    geometric_spreading: bool = False,
+):
+    """Generates an array of transmission loss factors due to geometrical spreading.
+
+    Args:
+        distances: An (n,m) np.array that contains the distance from nth source to mth
+            receiver.
+        geometric_spreading: If True this will apply loss from spherical spreading corresponding
```

Comment:
In the ideal case of plane wave propagation. This could be the case in wave guides, but for our use case this is not the case.
I only left this in to have an option to turn on/off the attenuation from geometric spreading to not have to deal with different amplitudes. As amplitude or received signal strength should not be a factor in what we are doing.